faiss/impl/ResidualQuantizer.cpp

```diff
@@ -328,22 +329,44 @@ void ResidualQuantizer::compute_codes(
         const float* x,
         uint8_t* codes_out,
         size_t n) const {
-    std::vector<float> residuals(max_beam_size * n * d);
-    std::vector<int32_t> codes(max_beam_size * M * n);
-    std::vector<float> distances(max_beam_size * n);
-
-    refine_beam(
-            n,
-            1,
-            x,
-            max_beam_size,
-            codes.data(),
-            residuals.data(),
-            distances.data());
-
-    // pack only the first code of the beam (hence the ld_codes=M *
-    // cur_beam_size)
-    pack_codes(n, codes.data(), codes_out, M * max_beam_size);
+    // split into chunks to avoid allocating too large memory
```

Comment:
could you please remove this chunking logic? 
I have a PR in the making that does exactly that.

Reply:
As long as https://github.com/facebookresearch/faiss/pull/1865 is merged, I will remove it and update this branch to make sure no conflicts.